src/puppet-web/firer.ts

```diff
@@ -315,22 +324,35 @@ async function checkRoomLeave(m: Message): Promise<void> {
   }
   /**
    * FIXME: leaver maybe is a list
+   * @lijiarui: I have checked, leaver will never be a list. If the bot remove 2 leavers at the same time, it will be 2 sys message, instead of 1 sys message contains 2 leavers.
    */
-  let leaverContact = room.member(leaver)
-
-  if (!leaverContact) {
-    log.error('PuppetWebFirer', 'fireRoomLeave() leaver %s not found, event `room-leave` & `leave` will not be fired', leaver)
-    return
+  let leaverContact: Contact | null, removerContact: Contact | null
+  if (leaver === this.userId) {
+    leaverContact = Contact.load(this.userId)
+    removerContact = room.member({alias: remover}) || room.member({name: remover})
```

Comment:
I think using `room.member()` is not strict here， because `roomAlias`(alias) here is useless, bot just recognize `contactAlias` and `name`, if we use `roomAlias`, then when someone's roomAlias is the same as other's name in the room. it will get the contact we don't mean go get.

So, maybe the new change I commit is the better way. we should may the code easier for developer, but more strict here.